### PR TITLE
Add a Scala SparkSink plugin type

### PR DESCRIPTION
**Pull request by albertshau**
Refactoring the code that compiles and executes scala code into
a common class, since the SparkSink and SparkCompute plugins
are exactly the same, except SparkCompute expects a RDD or Dataframe
to be returned, and SparkSink expects nothing to be returned.

### CODE REVIEW

File: src/main/java/co/cask/hydrator/plugin/spark/dynamic/ScalaSparkCodeExecutor.java [hunk 1]
```diff
@@ -0,0 +1,395 @@
...
+      if (sqlContext == null || sqlContext.sparkContext().isStopped()) {
+        sqlContext = new SQLContext(sc);
+        sqlContextThreadLocal.set(sqlContext);
+      }
+    }
+
+    return sqlContext;
+  }
+
+  private Method getMethod(ClassLoader classLoader, String className) {
```
Comment (albertshau) [inline src/main/java/co/cask/hydrator/plugin/spark/dynamic/ScalaSparkCodeExecutor.java:257]: this is the only part of this class that is not a direct move of code from ScalaSparkCompute. It has been modified to support a different method name and different return type.

File: src/main/java/co/cask/hydrator/plugin/spark/dynamic/ScalaSparkCodeExecutor.java [hunk 2]
```diff
@@ -0,0 +1,395 @@
...
+
+      // If it has second parameter, then must be SparkExecutionPluginContext
+      if (parameterTypes.length == 2 && !SparkExecutionPluginContext.class.equals(parameterTypes[1])) {
+        throw new IllegalArgumentException(String.format(
+          "The second parameter of the '%s' method should have type as SparkExecutionPluginContext", methodName));
+      }
+
+      // The return type of the method must be Unit
+      if (isVoid) {
+        if (!"void".equals(method.getReturnType().getName())) {
```
Comment (chtyim) [inline src/main/java/co/cask/hydrator/plugin/spark/dynamic/ScalaSparkCodeExecutor.java]: Better compare it with `void.class.equals(method.getReturnType())`

Comment (albertshau) [reply src/main/java/co/cask/hydrator/plugin/spark/dynamic/ScalaSparkCodeExecutor.java]: ah that's the class. Wasn't expecting all lowercase class.

**Comment by chtyim** [comment]
LGTM

**Pull request merged**